### PR TITLE
[Mellanox] Skip the leftover hardware reboot cause in case of last boot is warm/fast reboot

### PR DESCRIPTION
**Pull request by keboliu**
#### Why I did it
In case of warm/fast reboot, the hardware reboot cause will NOT be cleared because CPLD will not be touched in this flow. To not confuse the reboot cause determine logic, the leftover hardware reboot cause shall be skipped by the platform API, platform API will return the 'REBOOT_CAUSE_NON_HARDWARE'  instead of the "hardware" reboot cause.

#### How I did it
Check the proc cmdline to see whether the last reboot is a warm or fast reboot, if yes skip checking the leftover hardware reboot cause.

#### How to verify it
a. Manual test:
1. Perform a power loss
2. Perform a warm/fast reboot
3. check the reboot cause should be "warm-reboot" or "fast-reboot" instead of "power loss" 

b. Run reboot cause related regression test.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211

#### A picture of a cute animal (not mandatory but encouraged)

### CODE REVIEW

**Comment by prgeor** [comment]
@keboliu please captue the output of "show reboot-cause" and "show reboot-cause history" while doing this

1. Do warm-reboot
2. disconnect the power while system is coming up

**Review by stephenxs — APPROVED** [review]

File: platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py [hunk 1]
```diff
@@ -748,6 +765,14 @@ def get_reboot_cause(self):
             to pass a description of the reboot cause.
         """
         #read reboot causes files in the following order
+
+        # To avoid the leftover hardware reboot cause confusing the reboot cause determine service
+        # Skip the hardware reboot cause check if warm/fast reboot cause found from cmdline
+        if utils.is_host():
+            reboot_cause = self._parse_warmfast_reboot_from_proc_cmdline()
```
Comment (prgeor) [inline platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py:772]: reboot_cause will be None when system is booting after user issued "sudo reboot". ?

Comment (keboliu) [reply platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py:772]: New function _parse_warmfast_reboot_from_proc_cmdline() is designed to check the /proc/cmdline to see whether the last boot is a warm/fast reboot,  if Not, then the local variable 'reboot_cause' will be 'None', but this 'reboot_cause' is not the final reboot cause which will be returned by this platform API in this case, there will be further logic to determine the reboot cause from the sysfs if you check the reset of the code.

**Review by prgeor — APPROVED** [review]

**Comment by prgeor** [comment]
@lguohan could you help merge this change?

**Review by liat-grozovik — APPROVED** [review]

**Pull request merged**